sdk/servicebus/azure-servicebus/azure/servicebus/aio/async_message.py

```diff
@@ -89,16 +94,34 @@ async def dead_letter(self, description=None):
         To receive dead-lettered messages, use `QueueClient.get_deadletter_receiver()` or
         `SubscriptionClient.get_deadletter_receiver()`.
 
-        :param description: The reason for dead-lettering the message.
-        :type description: str
+        :param str description: The error description for dead-lettering the message.
+        :param str reason: The reason for dead-lettering the message. If `reason` is not set while `description` is
+         set, then `reason` would be set the same as `description`.
         :raises: ~azure.servicebus.common.errors.MessageAlreadySettled if the message has been settled.
         :raises: ~azure.servicebus.common.errors.MessageLockExpired if message lock has already expired.
         :raises: ~azure.servicebus.common.errors.SessionLockExpired if session lock has already expired.
         :raises: ~azure.servicebus.common.errors.MessageSettleFailed if message settle operation fails.
         """
         self._is_live('reject')
+
+        info = None
+        if description:
+            info = {
+                RECEIVER_LINK_DEAD_LETTER_REASON: reason if reason else description,
```

Comment:
```suggestion
                RECEIVER_LINK_DEAD_LETTER_REASON: reason or description,
```
Silly nit, wouldn't even suggest this if I couldn't make the change for you and have it be one click.  (Just some cute syntactic sugar for condensing that sort of ternary)